### PR TITLE
[executor][fix] Fix the bug of skipping a wrong number of txns when executing idempotent commit

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

#1627 introduces a bug that the skipping logic is not correct, which can cause more skipping than expect and failed the commits. Fixed in the PR

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Cluster_test

## Related PRs

#1627

### CODE REVIEW

**Comment by wqfish** [comment]
Test?

**Comment by msmouse** [comment]
@bors-libra delegate+

tests would be nice

**Comment by ankushagarwal** [comment]
FYI: We are crashlooping with this commit as well:

```
D1115 21:11:12.010241 139855547856640 consensus/src/chained_bft/event_processor.rs:327] Starting sync: current_hqc_round = 26072, sync_info_hqc_round = 26072, deadline = "1.491034858s"
D1115 21:11:12.010272 139855547856640 consensus/src/chained_bft/event_processor.rs:344] Caught up to HQC at round 26072
D1115 21:11:12.010322 139855547856640 executor/src/lib.rs:460] Received request to commit block 916a96a69c55a4c7c84f9d21dca26b34fb1a53948c57781d506d200690e2476a.
I1115 21:11:12.010345 139855547856640 executor/src/block_processor.rs:479] The lastest committed version: 1794007, the first version to commit: 1793908.Skipping the first 100 transactions and start committing from version 1794008
C1115 21:11:12.307996 139855547856640 common/crash-handler/src/lib.rs:36] details = '''
panicked at 'assertion failed: `(left == right)`
  left: `1794008`,
 right: `1794009`', executor/src/block_processor.rs:521:13'''
backtrace = '''
```

File: executor/src/block_processor.rs [hunk 1]
```diff
@@ -477,30 +477,35 @@ where
...
-                        txn_data.status().vm_status().major_status,
-                    ));
-                    num_accounts_created += txn_data.num_account_created();
-                }
+        for (txn, txn_data) in block_batch
+            .blocks
+            .iter()
+            .map(|block| itertools::zip_eq(&block.transactions, block.output.transaction_data()))
+            .flatten()
+            .skip(num_txns_to_skip as usize)
```
Comment (wqfish) [inline executor/src/block_processor.rs]: I'm not sure about this logic. You are skipping the first `num_txns_to_skip` transactions. Or do you want to skip the first `num_txns_to_skip` transactions whose status is `Keep`? Or something else? I just find it hard to understand.

Comment (lightmark) [reply executor/src/block_processor.rs]: There are more than 1 problem. Let me force-update a newer change.

**Pull request closed without merging**